### PR TITLE
Opensearch modules

### PR DESCRIPTION
**Pull request by chrisandreae**
Just clones of the ElasticSearch ones with names/binaries/environment variables updated.

- [x] Darwin
- [x] NixOS

### CODE REVIEW

File: nixos-modules/opensearch-dashboards.nix [hunk 1]
```diff
@@ -0,0 +1,164 @@
+{ config, lib, options, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.opensearch-dashboards;
+  opt = options.services.opensearch-dashboards;
+
+  ge7 = true;
```
Comment (thefloweringash) [inline nixos-modules/opensearch-dashboards.nix]: Unused?

File: nixos-modules/opensearch.nix [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+        while ! ${pkgs.curl}/bin/curl -sS -f http://${cfg.listenAddress}:${toString cfg.port} 2>/dev/null; do
+          sleep 1
+        done
+      '';
+    };
+
+    environment.systemPackages = [ cfg.package ];
+
+    users = {
+      groups.opensearch.gid = config.ids.gids.opensearch;
```
Comment (thefloweringash) [inline nixos-modules/opensearch.nix]: I don't think these ids are defined in nixpkgs. It's a [static list](https://github.com/NixOS/nixpkgs/blob/d4de0d56282716fcd54a077e80aaa41c567f83b4/nixos/modules/misc/ids.nix).

File: darwin-modules/opensearch-dashboards.nix [hunk 3]
```diff
@@ -0,0 +1,157 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.opensearch-dashboards;
+
+  ge7 = true;
```
Comment (thefloweringash) [inline darwin-modules/opensearch-dashboards.nix]: Also unused?

**Review by thefloweringash — COMMENTED** [review]
It all looks sensible to me. I don't know if the nixes-modules will evaluate due to the uid/gid though.

File: nixos-modules/opensearch-dashboards.nix (hunk 1 above)
Comment (chrisandreae) [reply nixos-modules/opensearch-dashboards.nix]: I didn't exactly write these, I made minimum changes to the upstream elasticsearch ones. But yeah, cutting things that aren't used seems great to me.

File: nixos-modules/opensearch.nix (hunk 2 above)
Comment (chrisandreae) [reply nixos-modules/opensearch.nix]: Oh. Should it steal the elasticsearch one then?

File: nixos-modules/opensearch-dashboards.nix (hunk 1 above)
Comment (thefloweringash) [reply nixos-modules/opensearch-dashboards.nix]: Indeed, but it looks like you also eliminated the usages.

File: nixos-modules/opensearch.nix (hunk 2 above)
Comment (thefloweringash) [reply nixos-modules/opensearch.nix]: That seems like a pragmatic solution. I don't have a better suggestion.

**Review by thefloweringash — APPROVED** [review]
Tested evaluation on aarch64-linux, LGTM.

**Pull request merged**